### PR TITLE
add test 'test_check_number_of_processes'

### PR DESCRIPTION
**Pull request by karolinavelkaja**
Signed-off-by: Karolina Rogowska <karolina.rogowska@intel.com>

### CODE REVIEW

**Comment by katlapinka** [comment]
retest this please

File: test/functional/tests/memory/test_check_number_of_processes.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        core_disk.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_disk.partitions[0]
+
+        cache = casadm.start_cache(cache_dev, force=True)
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Check number of IO threads."):
+        cmd = f"cat /proc/cpuinfo | grep processor | wc -l"
+        output = TestRun.executor.run(cmd).stdout
+        number_of_processors = int(output)
```
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:45]: tip: test_utils/os_utils

File: test/functional/tests/memory/test_check_number_of_processes.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        io_threads_actual = get_number_of_processes("cas_io")
+
+        validate_threads_number("IO threads", io_threads_actual, (number_of_processors + 1))
```
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:48]: why +1?

File: test/functional/tests/memory/test_check_number_of_processes.py (hunk 1 above)
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:45]: tip: test_utils/os_utils

File: test/functional/tests/memory/test_check_number_of_processes.py (hunk 2 above)
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:48]: why +1?

**Review by Ostrokrzew — APPROVED** [review]

File: test/functional/tests/memory/test_check_number_of_processes.py (hunk 1 above)
Comment (katlapinka) [reply test/functional/tests/memory/test_check_number_of_processes.py:45]: I also would like you to move this to os_utils

File: test/functional/tests/memory/test_check_number_of_processes.py (hunk 2 above)
Comment (karolinavelkaja) [reply test/functional/tests/memory/test_check_number_of_processes.py:48]: That was something from old test and I've checked that those 2 values differ by 1 indeed so for comparison the +1 is needed.

Comment (karolinavelkaja) [reply test/functional/tests/memory/test_check_number_of_processes.py:48]: additional info: processors from /proc/cpuinfo are zero-based which explains the 1 difference between values.

**Comment by karolinavelkaja** [comment]
Needs: https://github.com/Open-CAS/test-framework/pull/227/files

File: test/functional/tests/memory/test_check_number_of_processes.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Check number of IO threads."):
+        number_of_processors = get_number_of_processors_from_cpuinfo()
+        io_threads_actual = get_number_of_processes("cas_io")
+
+        validate_threads_number("IO threads", io_threads_actual, number_of_processors)
+
+    with TestRun.step("Check number of cleaning threads."):
+        cleaning_threads_actual = get_number_of_processes("cas_clean")
```
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:51]: Cleaner threads are named "cas_cl_%s".

File: test/functional/tests/memory/test_check_number_of_processes.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_check_number_of_processes():
+    """
+        title: Check the number of processes created by CAS.
+        description: |
+          For created CAS device check number of IO threads, cleaning threads, fill threads,
+          metadata updater threads.
```
Comment (Ostrokrzew) [inline test/functional/tests/memory/test_check_number_of_processes.py:27]: There's also management thread - 1 process for each cache.
name pattern: "cas_mngt_%s"

**Review by Ostrokrzew — CHANGES_REQUESTED** [review]

**Review by Ostrokrzew — APPROVED** [review]

**Review by katlapinka — APPROVED** [review]

**Pull request merged**